Send action events in the background

paulcsmith opened the pull request:
So that there isn't as much of a performance hit. For high throughput
applications we'll probably want to queue and batch requests, but this
is fine for most every application we work on.

Tests were written in a way that they pass with this change. Woo hoo!

SViccari commented on lib/reduxable_plug/background_job.ex:
```diff
@@ -0,0 +1,9 @@
+defmodule ReduxablePlug.BackgroundJob do
+  def send(fun) do
```
How does the TaskSupervisor know to be applied to anything passed to `send`?

paulcsmith replied:
Because I forgot to add it haha. Newb mistake :P

SViccari replied:
ah! okay, that makes a bit more sense now.

paulcsmith committed:
Send action events in the background

So that there isn't as much of a performance hit. For high throughput
applications we'll probably want to queue and batch requests, but this
is fine for most every application we work on.

paulcsmith commented:
Pretend you never saw the previous version @SViccari  :) This new one actually send in the background

paulcsmith commented on mix.exs:
```diff
@@ -15,7 +15,10 @@ defmodule ReduxablePlug.Mixfile do
-    [extra_applications: [:logger]]
+    [
+      extra_applications: [:logger],
+      mod: {ReduxablePlug.App, []}
```
This is how Elixir knows how to start the application

paulcsmith replied:
It looks for a `ReduxablePlug.App` module and calls it's `start` function with zero arguments (`[]`)

paulcsmith commented on lib/reduxable_plug/app.ex:
```diff
@@ -0,0 +1,14 @@
+    opts = [strategy: :one_for_one, name: ReduxablePlug.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+end
```
This is called when the app starts. It starts up a supervisor that monitors our background jobs

SViccari reviewed (approved):
Looks good! 🍺
